session/handler.go

```diff
@@ -349,6 +366,83 @@ func (h *Handler) adminListSessions(w http.ResponseWriter, r *http.Request, ps h
 	h.r.Writer().Write(w, r, sess)
 }
 
+// Session Get Request
+//
+// The request object for getting a session in an administrative context.
+//
+// swagger:parameters adminGetSession
+// nolint:deadcode,unused
+type adminGetSessionRequest struct {
+	// ExpandOptions is a comma seperated list of all properties that must be expanded in the Session. If no value is provided, the expandable properties are skipped.
+	//
+	// enum: ["Identity","Devices"]
+	// required: false
+	// in: query
+	ExpandOptions string `json:"expand"`
+
+	// ID is the session's ID.
+	//
+	// required: true
+	// in: path
+	ID string `json:"id"`
+}
+
+// swagger:route GET /admin/sessions/{id} v0alpha2 adminGetSession
+//
+// This endpoint returns the session object with expandables specified.
+//
+// This endpoint is useful for:
+//
+// - Getting a session object with all specified expandables that exist in an administrative context.
+//
+//	Schemes: http, https
+//
+//	Security:
+//	  oryAccessToken:
+//
+//	Responses:
+//	  200: session
+//	  400: jsonError
+//	  401: jsonError
+//	  404: jsonError
+//	  500: jsonError
+func (h *Handler) adminGetSession(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	if ps.ByName("id") == "whoami" {
+		// for /admin/sessions/whoami redirect to the public route
+		x.RedirectToPublicRoute(h.r)(w, r, ps)
```

Comment:
Is this a replacement for https://github.com/ory/kratos/pull/2855/files#diff-abfcc5544632668aab2f1162de82ec56e8ec7444f6ce377cb3a30a9eafbed198L75 ?

Reply:
Yes, the router panics on adding a `/admin/sessions/whoami` because it can't tell it apart from `/admin/sessions/<some string>` and this is a workaround to deal with that